src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -27,7 +48,13 @@ flex_image get_image(const flexible_type& image_feature) {
   }
 }
 
-gl_sframe get_data(const data_iterator::parameters& params) {
+gl_sframe get_data(data_iterator::parameters params) {
+  // Check annotations type. If it is a single dictionary, put it into a list.
+  if (params.data.contains_column(params.annotations_column_name)) {
+    gl_sarray raw_annotations = params.data[params.annotations_column_name];
+    params.data.replace_add_column(canonicalize_annotations(raw_annotations),
```

Comment:
The rest of the code in `get_data` takes some care not to mutate the user's original SFrame. Could we integrate this below, so that we don't call destructive (mutating) methods on the input SFrame, such as `replace_add_column`?